### PR TITLE
Ensure that environment headers & proxy headers are returned in respo…

### PR DESCRIPTION
**Pull request by abiodunjames**
**Description**
This PR ensures that  headers defined in proxy mode are always returned in responses.

**Related Issue**

https://github.com/mockoon/mockoon/issues/136

**Motivation and Context**
In my projects, I often use custom headers like `x-auth-key` for authentication. If you use Mockoon with this header (x-auth-key) set in an environment, it will work if the proxied server returns a  200 response.  Incase of a 404 error from the proxied server, the header `x-auth-key` is ignored. See https://github.com/mockoon/mockoon/issues/136

**How Has This Been Tested?**

* Use Mockoon in proxy mode with some headers defined. 
* Access some valid endpoints in the proxied server from your browser and inspect  the response headers. It should return the headers defined
* Access an endpoint that does not exist in the proxied server such that you get a 404 error.  The headers you defined in mockoon should be returned in the response as well.

** Unit/UI test?**
No, this is my entry PR to this project. Things are still sketchy for me.

### CODE REVIEW

**Comment by 255kb** [comment]
@abiodunjames Thank you for the update.
It could be great to add automated tests (there are all in the `test` folder). 
You could take example on https://github.com/mockoon/mockoon/blob/master/test/basic-calls.spec.ts and/or https://github.com/mockoon/mockoon/blob/master/test/proxy.spec.ts
I would call them something like `headers.spec.ts` as there is plenty to test (not only for your contribution). You can prepare the `environments.json` file like for other test suites, with existing routes, etc. 
Idea would be to test:
- simple call to non-existing route (to get an ExpressJS' 404, not a Mockoon declared 404) without a proxy, and see if environment headers are sent
- call to non-existing route on an env that proxy to another one and see if we still get initial environment headers

Does it make sense? Of course if you see other test cases you can implement them. If you are lost or don't have enough time, I can also write them :smiley:

File: src/app/services/server.service.ts [hunk 1]
```diff
@@ -300,22 +318,20 @@ export class ServerService {
       // logging the proxied response
       const self = this;
       const processResponse = (proxyRes, req, res) => {
+        const combinedHeaders = {...res.getHeaders(), ...proxyRes.headers};
```
[the remark this answers is not in the document]
Comment (255kb) [reply src/app/services/server.service.ts]: That make sense. Don't worry I will also retest everything, to ensure that there is no regression.

**Comment by 255kb** [comment]
Concerning the header presence assertion, could you please follow the way it's done in other tests? 
See an example there: https://github.com/mockoon/mockoon/blob/master/test/basic-calls.spec.ts#L11
`testeProperties` would need to be update in order to be able to assert headers.

**Comment by 255kb** [comment]
@abiodunjames Hi, I took some time off the project (mainly due to sicknesses 😆 ). I will start testing this, and if you agree I will take it from here, so I also write headers related tests (not only for this PR) that are missing anyway.

**Comment by 255kb** [comment]
@abiodunjames  I had to slightly modify your PR as proxy response headers were sent even when proxy mode was disabled, and normal 404 (without any proxy #136) weren't returning the environment's global headers. 
Apart from this, thanks for the great contribution! It will be in v1.8.0 (soon) :rocket:

**Comment by abiodunjames** [comment]
> @abiodunjames Hi, I took some time off the project (mainly due to sicknesses 😆 ). I will start testing this and will take it from here, so I also write headers related tests (not only for this PR) that are missing anyway.

Thanks for the change. Wow wow.. 🚀 v1.80

**Comment by 255kb** [comment]
@abiodunjames coming soon :)

**Pull request merged**